src/Sia.Gateway/Controllers/EventsController.cs

```diff
@@ -28,9 +30,11 @@ public class EventsController : BaseController
         }
 
         [HttpGet(Name = nameof(GetEvents))]
-        public async Task<IActionResult> GetEvents([FromRoute]long incidentId, [FromQuery]PaginationMetadata pagination)
+        public async Task<IActionResult> GetEvents([FromRoute]long incidentId,
+            [FromQuery]PaginationMetadata pagination,
+            [FromQuery]EventFilters filter)
         {
-            var result = await _mediator.Send(new GetEventsRequest(incidentId, pagination, _authContext));
+            var result = await _mediator.Send(new GetEventsRequest(incidentId, pagination, filter, _authContext));
             Response.Headers.AddPagination(new LinksHeader(pagination, _urlHelper, nameof(GetEvents)));
```

Comment:
We should call AddPagination passing a new FilteredLinksHeader. I think there may still be things we need to change to ensure that the next page and previous page links include the filter values.